embedded-keycloak/src/main/java/com/playtika/test/keycloak/KeycloakContainer.java

```diff
@@ -0,0 +1,90 @@
+package com.playtika.test.keycloak;
+
+import static com.playtika.test.common.utils.ContainerUtils.containerLogsConsumer;
+import static java.lang.String.format;
+import static java.time.Duration.ofSeconds;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.containers.wait.strategy.WaitStrategy;
+import org.testcontainers.utility.MountableFile;
+
+@Slf4j
+public class KeycloakContainer extends GenericContainer<KeycloakContainer> {
+
+    private static final int DEFAULT_HTTP_PORT_INTERNAL = 8080;
+    private static final String AUTH_BASE_PATH = "/auth";
+
+    private final KeycloakProperties properties;
+    private final ResourceLoader resourceLoader;
+
+    public KeycloakContainer(KeycloakProperties properties,
+        ResourceLoader resourceLoader) {
+        super(properties.getDockerImage());
+        this.properties = properties;
+        this.resourceLoader = resourceLoader;
+    }
+
+    @Override
+    protected void configure() {
+        withEnv("KEYCLOAK_HTTP_PORT", String.valueOf(DEFAULT_HTTP_PORT_INTERNAL));
+        withEnv("KEYCLOAK_USER", properties.getAdminUser());
+        withEnv("KEYCLOAK_PASSWORD", properties.getAdminPassword());
+        withCommand(properties.getCommand());
+        withStartupTimeout(properties.getTimeoutDuration());
+        withLogConsumer(containerLogsConsumer(log));
+        withExposedPorts(DEFAULT_HTTP_PORT_INTERNAL);
+        waitingFor(authBasePath());
+
+        String importFile = properties.getImportFile();
```

Comment:
better to have separate method ```configureKeyCloadImport``` or something, and keep this method cleaner and more readable